libs/ha/ha_common_libs_k8s.py

```diff
@@ -1480,3 +1480,106 @@ def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
+                        more_results = self.get_replace_recursively(item, field, replace_key,
+                                                                    replace_val)
+                        for another_result in more_results:
+                            fields_found.append(another_result)
+
+        return fields_found
+
+    def update_deployment_yaml(self, pod_obj, pod_name, find_key, replace_key=None,
+                               replace_val=None):
+        """
+        Function to find and replace key from deployment yaml file
+        :param pod_obj: Object of pod
+        :param pod_name: Name of the pod
+        :param find_key: Key to be searched
+        :param replace_key: Key with which older key to be replaced
+        :param replace_val: Value with which older value to be replaced
+        :return: bool, str, str (status, new path, backup path)
+        """
+        resp = pod_obj.get_deploy_replicaset(pod_name)
+        deploy = resp[1]
+        LOGGER.info("Deployment for pod %s is %s", pod_name, deploy)
+        LOGGER.info("Taking deployment backup")
+        resp = pod_obj.backup_deployment(deploy)
+        yaml_path = resp[1]
+        LOGGER.info("Copy deployment yaml file to local system")
+        pod_obj.copy_file_to_local(remote_path=yaml_path, local_path=yaml_path)
+        resp = config_utils.read_yaml(yaml_path)
+        if not resp[0]:
+            return resp
+        data = resp[1]
+        system_utils.remove_file(yaml_path)
+        LOGGER.info("Update %s of deployment yaml with %s : %s", find_key, replace_key, replace_val)
+        val = self.get_replace_recursively(data, find_key, replace_key, replace_val)
+        if not val:
+            return False, f"Failed to find key {find_key} in deployment yaml {yaml_path} of pod " \
+                          f"{pod_name}"
+        resp = config_utils.write_yaml(fpath=yaml_path, write_data=data, backup=True,
+                                       sort_keys=False)
+        LOGGER.debug("Response: %s", resp)
+        if not resp[0]:
+            return resp
+        LOGGER.info("Copying files %s and %s to master node", yaml_path, f'{yaml_path}.bkp')
+        resp = pod_obj.copy_file_to_remote(local_path=yaml_path, remote_path=yaml_path)
+        if not resp[0]:
+            return resp
+        resp = pod_obj.copy_file_to_remote(local_path=f'{yaml_path}.bkp',
+                                           remote_path=f'{yaml_path}.bkp')
+        if not resp[0]:
+            return resp
+
+        LOGGER.info("Successfully updated deployment yaml file for pod %s", pod_name)
+        return True, yaml_path, f'{yaml_path}.bkp'
+
+    @staticmethod
+    def failover_pod(pod_obj, pod_list, failover_node):
+        """
+        Function to failover given pod
+        :param pod_obj: Object of the pod
+        :param pod_list: List of the pods to be failed over
+        :param failover_node: Node to which pod is to be failed over
```

Comment:
so this failover node will be sent from test and what will be basis for the same?

Reply:
Yes, we need to send it from test. It can be any node, other than currently hosting node.